### PR TITLE
Add doc for JEP390 command line options

### LINKED ISSUE #665: New command line options for JEP390
**Issue or pull request number:**
Issue: https://github.com/eclipse/openj9/issues/10620
PR: https://github.com/eclipse/openj9/pull/11035, https://github.com/eclipse/openj9/pull/11432

**Overview:**
1. A new command line option to throw `VirtualMachineError` on improper attempts to synchronize on instances of value-based classes. 
2. A new command line option to warn user on improper attempts to synchronize on instances of value-based classes.

**Release target:**
Eclipse OpenJ9 0.25.0

**Applies to the following JDK versions:**
Java 16 and up

**Applies to the following platforms:**
All platforms

**For new command line options:**

- Option name and syntax: 
```
-XX:DiagnoseSyncOnValueBasedClasses=1
-XX:DiagnoseSyncOnValueBasedClasses=2
```
- If values are set:  
	N.A.
- Purpose:
1. With `-XX:DiagnoseSyncOnValueBasedClasses=1`
VM throws `VirtualMachineError` on improper attempts to synchronize on instances of value-based classes.
2. With `-XX:DiagnoseSyncOnValueBasedClasses=2`
VM prints out a warning message on improper attempts to synchronize on instances of value-based classes.
- Restrictions:
N.A.

### PR DESCRIPTION
**Pull request by hangshao0**
closes #665

Signed-off-by: Hang Shao <hangshao@ca.ibm.com>

### CODE REVIEW

**Comment by SueChaplain** [comment]
Jenkins doc stage

File: docs/xxdiagnosesynconvaluebasedclasses.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+-->
+
+# -XX:DiagnoseSyncOnValueBasedClasses
+
+![Start of content that applies only to Java 16 and later](cr/java16plus.png) This option provides warnings about improper attempts to synchronize on instances of a value-based class. ![End of content that applies only to Java 16 or later](cr/java_close.png)
+
+
+## Syntax
+
+        -XX:DiagnoseSyncOnValueBasedClasses=1
```
Comment (SueChaplain) [inline docs/xxdiagnosesynconvaluebasedclasses.md:32]: For consistency with the rest of the topics, please can we use the format:

```
-XX:DiagnoseSyncOnValueBasedClasses=<number>
```
Where `<number>` is one of the following values:

- `1`: OpenJ9 throws a `VirtualMachineError` on improper attempts to synchronize on instances of value-based classes.
- `2`: OpenJ9 prints ~~out~~ a warning message on improper attempts to synchronize on instances of value-based classes.

Comment (SueChaplain) [reply docs/xxdiagnosesynconvaluebasedclasses.md:32]: Is there a default?

**Comment by SueChaplain** [comment]
In addition to the comments for the -XX topic, please can you add the appropriate entries to the release topic `version025.md`.

File: docs/xxdiagnosesynconvaluebasedclasses.md (hunk 1 above)
Comment (hangshao0) [reply docs/xxdiagnosesynconvaluebasedclasses.md:32]: > Is there a default?

No, there is no default. User has to specify either 1 or 2 to `<number>`

Comment (hangshao0) [reply docs/xxdiagnosesynconvaluebasedclasses.md]: Updated according to the review comment.

**Comment by hangshao0** [comment]
> In addition to the comments for the -XX topic, please can you add the appropriate entries to the release topic version025.md.

Done. 

We are using the same option `-XX:DiagnoseSyncOnValueBasedClasses=1/2` introduced by JEP390 as hotspot. In `xxdiagnosesynconvaluebasedclasses.md`, not sure if the current description is fine, or is it better to say something like "This Oracle Hotspot option provides warnings about improper attempts to synchronize on instances of a value-based class. This option is recognized by OpenJ9 and provided for compatibility" ?

**Comment by SueChaplain** [comment]
Jenkins doc stage

**Comment by SueChaplain** [comment]
@hangshao0 - there are a few editorial changes I would make to align this with other topics that implement HotSpot options, but I'll merge this as-is and fix it up in another PR. Thanks for your help getting it this far.

**Pull request merged**